presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/ConnectionFactory.java

```diff
@@ -17,7 +17,12 @@
 import java.sql.SQLException;
 
 public interface ConnectionFactory
+        extends AutoCloseable
 {
     Connection openConnection()
             throws SQLException;
+
+    @Override
+    void close()
```

Comment:
Add a default implementation that does nothing and mark the interface as functional.